src/Lykke.Messaging/ProcessingGroupManager.cs

```diff
@@ -178,27 +174,19 @@ private string GetSessionName(ProcessingGroup processingGroup, int priority)
 
         private ProcessingGroup GetProcessingGroup(string processingGroup)
         {
-            ProcessingGroup group;
-            lock (m_ProcessingGroups)
-            {
-                if (m_ProcessingGroups.TryGetValue(processingGroup, out group)) 
-                    return group;
+            if (m_ProcessingGroups.TryGetValue(processingGroup, out var group))
+                return group;
+
+            var info = m_ProcessingGroupInfos.GetOrAdd(processingGroup, new ProcessingGroupInfo());
```

Comment:
Is ProcessingGroupInfo instantiation heavy operation?

Reply:
No